sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/PartitionProcessorEventsManager.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Azure.Messaging.EventHubs;
+
+namespace Microsoft.Azure.WebJobs.EventHubs.Listeners
+{
+    /// <summary>
+    /// This class manages a queue of event data using a minimum and maximum batch size. This class is NOT thread safe.
+    /// Concurrency must be managed by the class using this events manager.
+    /// </summary>
+    internal class PartitionProcessorEventsManager : IDisposable
+    {
+        private int _maxBatchSize;
+        private int _minBatchSize;
+
+        // This is internal for mocking purposes only.
+        internal Queue<EventData> CachedEvents { get; set; }
+
+        public bool HasCachedEvents
+        {
+            get
+            {
+                return CachedEvents.Count > 0;
+            }
+        }
+
+        public PartitionProcessorEventsManager(int maxBatchSize, int minBatchSize)
+        {
+            CachedEvents = new Queue<EventData>();
```

Comment:
nit: this might yield a slight perf benefit

```suggestion
            CachedEvents = new Queue<EventData>(maxBatchSize);
```

Reply:
Actually, maybe not - for large batch sizes this could be wasteful.